src/marqo/client.py

```diff
@@ -232,59 +236,76 @@ def health(self):
         ", instead use 'client.index(index_name).eject_model()"
     )
     def eject_model(self, model_name: str, model_device: str):
+        if isinstance(self.config.instance_mapping, MarqoCloudInstanceMappings):
+            self.raise_error_for_cloud()
         return self.http.delete(path=f"models?model_name={model_name}&model_device={model_device}")
 
     @deprecated(
         "This method is deprecated and will be removed in Marqo 2.0.0"
         ", instead use 'client.index(index_name).get_loaded_models()"
     )
     def get_loaded_models(self):
+        if isinstance(self.config.instance_mapping, MarqoCloudInstanceMappings):
+            self.raise_error_for_cloud()
         return self.http.get(path="models")
 
     @deprecated(
         "This method is deprecated and will be removed in Marqo 2.0.0"
         ", instead use 'client.index(index_name).get_cuda_info()"
     )
     def get_cuda_info(self):
+        if isinstance(self.config.instance_mapping, MarqoCloudInstanceMappings):
+            self.raise_error_for_cloud()
         return self.http.get(path="device/cuda")
 
     @deprecated(
         "This method is deprecated and will be removed in Marqo 2.0.0"
         ", instead use 'client.index(index_name).get_cpu_info()"
     )
     def get_cpu_info(self):
+        if isinstance(self.config.instance_mapping, MarqoCloudInstanceMappings):
+            self.raise_error_for_cloud()
         return self.http.get(path="device/cpu")
 
     def _marqo_minimum_supported_version_check(self):
         min_ver = minimum_supported_marqo_version()
+        url = self.config.instance_mapping.get_control_url()
```

Comment:
This won't work because the control plane won't serve this endpoint. Different data planes could be running different versions so this is a tricky one. I'll get back to you on this.